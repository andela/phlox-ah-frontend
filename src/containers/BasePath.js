export * from '../requests/LoginRequests';
export * from '../util/AsyncUtil';
export * from '../actionTypes/UserConstants';
export * from '../actions/MsgInfoActions';
export * from '../components/Input/Input';
export * from '../components/Button/Button';
export * from '../requests/SignupRequests';
export * from '../requests/ArticleRequests';
export * from '../requests/ProfileRequest';

