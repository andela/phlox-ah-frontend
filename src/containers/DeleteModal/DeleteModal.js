import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Preloader } from 'react-materialize';
import { deleteArticle } from '../../requests/ArticleRequests';

import './DeleteModal.scss';

class DeleteModal extends Component {
  constructor() {
    super();

    this.deleteArticle = this.deleteArticle.bind(this);
    this.hideDeleteModal = this.hideDeleteModal.bind(this);
  }

  hideDeleteModal() {
    if (this.props.deleteItem.loading) return;
    this.props.hideDeleteModal();
  }

  deleteArticle() {
    if (this.props.deleteItem.loading) return;
    console.log(this.props.articleslug);
    this.props.deleteArticle(this.props.articleslug)
      .then(this.hideDeleteModal);
  }

  render() {
    return (
      <div className="delete-modal">
        <div className="modal-body">
          {
            this.props.deleteItem.loading
              && <div className="loader-wrapper">
                <Preloader />
              </div>
          }
          <div className="modal-info">
            <span className="warning-icon">
              <i
                className="fas fa-exclamation-triangle">
              </i>
            </span>
            <span className="warning-text">
              Are you sure you want to delete this article?
            </span>
          </div>
          <div className="modal-buttons">
            <button
              onClick={this.deleteArticle}
              className="btn yes">
              Yes
            </button>
            <button
              onClick={this.hideDeleteModal}
              className="btn no">
              No
            </button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  deleteItem: state.deleteItem
});

export default connect(mapStateToProps, {
  deleteArticle
})(DeleteModal);
