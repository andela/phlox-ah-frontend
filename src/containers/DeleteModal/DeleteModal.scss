div.delete-modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0,0,0,.5);
  z-index: 1100;
  display: flex;

  div.modal-body {
    margin: auto;
    min-width: 300px;
    width: 30%;
    background: #fff;
    height: auto;
    padding: 20px;
    border-radius: 3px;
    position: relative;

    div.loader-wrapper {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }

    div.modal-info {
      display: flex;
      color: #444;
      margin: 10px;
      align-items: center;
    }

    div.modal-info span{display: block;}

    div.modal-info span.warning-icon {
      font-size: 3em;
      color: red;
      margin-right: 15px;
      display: block;
    }

    div.modal-info span.warning-text {
      font-size: 20px;
      font-weight: 600;
      display: block;
      ;
    }

    div.modal-buttons {
      margin-top: 30px;
      margin-bottom: 0px;
      margin-left: auto;
      margin-right: auto;
      width: 80%;
      text-align: center;
    }

    div.modal-buttons button {
      width: 90px;
      font-weight: 600;
    }

    div.modal-buttons button.yes {
      background: #cc5b5b;
      margin-right: 10px;
    }

  }
}
