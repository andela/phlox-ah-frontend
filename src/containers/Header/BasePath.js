export * from '../../actionTypes/UserConstants';
export * from '../../util/AsyncUtil';
