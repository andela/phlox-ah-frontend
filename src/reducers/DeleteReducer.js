import { DELETE } from '../actionTypes/DeleteContants';
import { asyncActionName } from '../util/AsyncUtil';


const initialState = {
  loading: false
};

const deleteReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case asyncActionName(DELETE).loading:
      return { ...state, loading: action.payload };
    default:
      return state;
  }
};

export default deleteReducer;
