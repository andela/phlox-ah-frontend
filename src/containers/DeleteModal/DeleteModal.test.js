import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import DeleteModal from './DeleteModal';
import DeleteReducer from '../../reducers/DeleteReducer';
import { asyncActions, DELETE } from '../BasePath';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  deleteItem: {
    loading: true,
  },
  articleslug: 'title-slug',
  hideDeleteModal: jest.fn(),
  deleteArticle: jest.fn()
});

let component;
let myComponent;

describe('<DeleteModal />', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <DeleteModal />
      </Provider>
    );
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have span.warning-icon', () => {
    expect(myComponent.find('span.warning-icon').exists()).toBe(true);
  });
  it('should have div.modal-info', () => {
    expect(myComponent.find('div.modal-info').exists()).toBe(true);
  });
  it('should have div.modal-buttons', () => {
    expect(myComponent.find('div.modal-buttons').exists()).toBe(true);
  });
  it('should have span.warning-text', () => {
    expect(myComponent.find('span.warning-text').exists()).toBe(true);
  });
  it('should have button.btn.yes tag', () => {
    expect(myComponent.find('button.btn.yes').exists()).toBe(true);
  });
  it('should have button.btn.no tag', () => {
    expect(myComponent.find('button.btn.no').exists()).toBe(true);
  });
  it('should have div.delete-modal', () => {
    expect(myComponent.find('div.delete-modal').exists()).toBe(true);
  });
  it('should have Preloader component', () => {
    expect(myComponent.find('Preloader').exists()).toBe(true);
  });
  it('should have deleteItem.loading of true', () => {
    expect(myComponent.instance().props.deleteItem.loading).toBe(true);
  });
  it('should have hideDeleteModal function', () => {
    expect(myComponent.instance().hideDeleteModal).toBeDefined();
  });
  it('should have deleteArticle prop function', () => {
    expect(myComponent.instance().props.deleteArticle).toBeDefined();
  });
  it('should create an action to set loading to false', () => {
    const payload = {
      loading: false
    };
    const expectedAction = {
      type: 'DELETE_LOADING',
      payload
    };
    expect(asyncActions(DELETE).loading(payload)).toEqual(expectedAction);
    const newState = DeleteReducer({}, expectedAction);
    expect(newState).toEqual({ loading: payload });
  });
  it('should create an action to set loading to true', () => {
    const payload = {
      loading: true
    };
    const expectedAction = {
      type: 'DELETE_LOADING',
      payload
    };
    expect(asyncActions(DELETE).loading(payload)).toEqual(expectedAction);
    const newState = DeleteReducer({}, expectedAction);
    expect(newState).toEqual({ loading: payload });
  });
});
