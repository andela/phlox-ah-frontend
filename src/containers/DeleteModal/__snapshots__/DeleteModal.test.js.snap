// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteModal /> should render without throwing an error 1`] = `<Connect(DeleteModal) />`;
